### ISSUE TITLE
`The UNC path should be of the form \\server\share` when importing directly from UNC shared folder

### ISSUE THREAD

**Issue by manniebugs** [Sonarr/Sonarr#4943]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

**TLDR**; During Import of a file, Flood Download Client implementation has a problem with UNC paths when importing single episode downloads. 

**UPDATE**: workaround in the comments.

When trying to import an single episode either manually or automatically after a download has completed it will fail with the exception:
```code
2022-03-14 18:01:35.3|Warn|ImportApprovedEpisodes|Couldn't import episode \\1.1.1.1\downloads\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv

[v3.0.7.1477] System.ArgumentException: The UNC path should be of the form \\server\share.
   at System.IO.LongPathHelper.Normalize(String path, UInt32 maxPathLength, Boolean checkInvalidCharacters, Boolean expandShortPaths)
   at System.IO.Path.NewNormalizePath(String path, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.GetFullPathInternal(String path)
   at System.IO.DirectoryInfo.Init(String path, Boolean checkHost)
   at System.IO.DirectoryInfo..ctor(String path)
   at NzbDrone.Common.Extensions.PathExtensions.IsParentPath(String parentPath, String childPath) in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Common\Extensions\PathExtensions.cs:line 116
   at NzbDrone.Core.MediaFiles.EpisodeImport.ImportApprovedEpisodes.GetOriginalFilePath(DownloadClientItem downloadClientItem, LocalEpisode localEpisode) in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\MediaFiles\EpisodeImport\ImportApprovedEpisodes.cs:line 169
   at NzbDrone.Core.MediaFiles.EpisodeImport.ImportApprovedEpisodes.Import(List`1 decisions, Boolean newDownload, DownloadClientItem downloadClientItem, ImportMode importMode) in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\MediaFiles\EpisodeImport\ImportApprovedEpisodes.cs:line 107

```
This **does not occur** if the episode is in a folder or a season pack. This also has only started happening when switching to the Flood download client. This behaviour is not present with other tested clients (Deluge & rTorrent).

### Expected Behavior

The episode would be imported to Sonar and copied to the desired folder.

### Steps To Reproduce

1. **Enviroment**
Sonar running on Windows 10. Flood / rTorrent running on Ubuntu.
2. **Config**
```code
rTorrent download folder: /mnt/downloads

Sonar Remote Path Mappings:
Host: Flood
Remote Path: /mnt/downloads
Local Path: \\1.1.1.1\downloads
```
3. **Action**
Running Manually import downloads `(ImportApprovedEpisodes)` or after an episode has finished downloading 
4. **Error Event**
```code
Component               | Message
ImportApprovedEpisodes	| Couldn't import episode \\1.1.1.1\downloads\Episode.mkv: The UNC path should be of the form \\server\share.
```

### Environment

```markdown
Sonarr Environment:
- OS: Windows 10 (21H2)
- Sonarr: 3.0.7.1477
- Docker Install: No
- Using Reverse Proxy: No
- Browser: Chrome / Safari

Flood Environment:
- Flood (4.7.0) / rTorrent (0.9.8)
- Ubuntu 20.04 server
```

### What branch are you running?

Main

### Trace Logs?

```code
2022-03-14 18:01:33.9|Trace|Http|Req: 2236 [GET] /api/v3/manualimport?downloadId=FFEF1D8B1B7ACB3D57FD5D6FAF17249960B596A4&folder=&filterExistingFiles=true (from ::1 Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36 Edg/99.0.1150.39)
2022-03-14 18:01:33.9|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv'
2022-03-14 18:01:33.9|Trace|Parser|^\[(?<subgroup>.+?)\][-_. ]?(?<title>[^-]+?)(?:(?<![-_. ]|\b[0]\d+) - )(?:[-_. ]?(?<absoluteepisode>\d{2,3}(\.\d{1,2})?(?!\d+)))+(?:[-_. ]+(?<special>special|ova|ovd))?.*?(?<hash>\[\w{8}\])?(?:$|\.mkv)
2022-03-14 18:01:33.9|Debug|Parser|Episode Parsed. Shingeki no Kyojin (The Final Season) - 085 
2022-03-14 18:01:33.9|Debug|Parser|Language parsed: English
2022-03-14 18:01:34.0|Debug|QualityParser|Trying to parse quality for [SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.0|Debug|Parser|Quality parsed: HDTV-720p v1
2022-03-14 18:01:34.0|Debug|Parser|Release Group parsed: SubsPlease
2022-03-14 18:01:34.0|Debug|Parser|Release Hash parsed: 43E56853
2022-03-14 18:01:34.0|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]'
2022-03-14 18:01:34.0|Debug|SceneMappingService|Found scene mapping for: Shingeki no Kyojin (The Final Season). TVDB ID for mapping: 267440
2022-03-14 18:01:34.0|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv'
2022-03-14 18:01:34.0|Trace|Parser|^\[(?<subgroup>.+?)\][-_. ]?(?<title>[^-]+?)(?:(?<![-_. ]|\b[0]\d+) - )(?:[-_. ]?(?<absoluteepisode>\d{2,3}(\.\d{1,2})?(?!\d+)))+(?:[-_. ]+(?<special>special|ova|ovd))?.*?(?<hash>\[\w{8}\])?(?:$|\.mkv)
2022-03-14 18:01:34.0|Debug|Parser|Episode Parsed. Shingeki no Kyojin (The Final Season) - 085 
2022-03-14 18:01:34.0|Debug|Parser|Language parsed: English
2022-03-14 18:01:34.0|Debug|QualityParser|Trying to parse quality for [SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.0|Debug|Parser|Quality parsed: HDTV-720p v1
2022-03-14 18:01:34.0|Debug|Parser|Release Group parsed: SubsPlease
2022-03-14 18:01:34.0|Debug|Parser|Release Hash parsed: 43E56853
2022-03-14 18:01:34.0|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]'
2022-03-14 18:01:34.0|Debug|SceneMappingService|Found scene mapping for: Shingeki no Kyojin (The Final Season). TVDB ID for mapping: 267440
2022-03-14 18:01:34.0|Debug|ImportDecisionMaker|Analyzing 1/1 files.
2022-03-14 18:01:34.0|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv'
2022-03-14 18:01:34.0|Trace|Parser|^\[(?<subgroup>.+?)\][-_. ]?(?<title>[^-]+?)(?:(?<![-_. ]|\b[0]\d+) - )(?:[-_. ]?(?<absoluteepisode>\d{2,3}(\.\d{1,2})?(?!\d+)))+(?:[-_. ]+(?<special>special|ova|ovd))?.*?(?<hash>\[\w{8}\])?(?:$|\.mkv)
2022-03-14 18:01:34.0|Debug|Parser|Episode Parsed. Shingeki no Kyojin (The Final Season) - 085 
2022-03-14 18:01:34.0|Debug|Parser|Language parsed: English
2022-03-14 18:01:34.0|Debug|QualityParser|Trying to parse quality for [SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.0|Debug|Parser|Quality parsed: HDTV-720p v1
2022-03-14 18:01:34.0|Debug|Parser|Release Group parsed: SubsPlease
2022-03-14 18:01:34.0|Debug|Parser|Release Hash parsed: 43E56853
2022-03-14 18:01:34.0|Debug|VideoFileInfoReader|Getting media info from \\1.1.1.1\downloads\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.1|Trace|MediaInfo|Read file offset 0-5898240 (5898240 bytes)
2022-03-14 18:01:34.1|Trace|MediaInfo|Read file offset 745804813-745826220 (21407 bytes)
2022-03-14 18:01:34.1|Trace|MediaInfo|Read a total of 5919647 bytes (0.8%)
2022-03-14 18:01:34.1|Debug|DetectSample|Runtime is over 90 seconds
2022-03-14 18:01:34.1|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv'
2022-03-14 18:01:34.1|Trace|Parser|^\[(?<subgroup>.+?)\][-_. ]?(?<title>[^-]+?)(?:(?<![-_. ]|\b[0]\d+) - )(?:[-_. ]?(?<absoluteepisode>\d{2,3}(\.\d{1,2})?(?!\d+)))+(?:[-_. ]+(?<special>special|ova|ovd))?.*?(?<hash>\[\w{8}\])?(?:$|\.mkv)
2022-03-14 18:01:34.1|Debug|Parser|Episode Parsed. Shingeki no Kyojin (The Final Season) - 085 
2022-03-14 18:01:34.1|Debug|Parser|Language parsed: English
2022-03-14 18:01:34.1|Debug|QualityParser|Trying to parse quality for [SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.1|Debug|Parser|Quality parsed: HDTV-720p v1
2022-03-14 18:01:34.1|Debug|Parser|Release Group parsed: SubsPlease
2022-03-14 18:01:34.1|Debug|Parser|Release Hash parsed: 43E56853
2022-03-14 18:01:34.1|Debug|VideoFileInfoReader|Getting media info from \\1.1.1.1\downloads\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.2|Trace|MediaInfo|Read file offset 0-5898240 (5898240 bytes)
2022-03-14 18:01:34.2|Trace|MediaInfo|Read file offset 745804813-745826220 (21407 bytes)
2022-03-14 18:01:34.2|Trace|MediaInfo|Read a total of 5919647 bytes (0.8%)
2022-03-14 18:01:34.2|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]'
2022-03-14 18:01:34.2|Debug|SceneMappingService|Found scene mapping for: Shingeki no Kyojin (The Final Season). TVDB ID for mapping: 267440
2022-03-14 18:01:34.2|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]'
2022-03-14 18:01:34.2|Debug|SceneMappingService|Found scene mapping for: Shingeki no Kyojin (The Final Season). TVDB ID for mapping: 267440
2022-03-14 18:01:34.2|Debug|ParsingService|Using absolute episode number 85 for: Attack on Titan - TVDB: 4x26
2022-03-14 18:01:34.2|Debug|AggregateLanguage|Using language: English
2022-03-14 18:01:34.2|Trace|PreferredWordService|Calculating preferred word score for '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]'
2022-03-14 18:01:34.2|Trace|PreferredWordService|Calculated preferred word score for '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]': 0 (0 match(es))
2022-03-14 18:01:34.2|Trace|PreferredWordService|Calculating preferred word score for '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]'
2022-03-14 18:01:34.2|Trace|PreferredWordService|Calculated preferred word score for '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]': 0 (0 match(es))
2022-03-14 18:01:34.2|Trace|AggregateQuality|Considering Source Television (Fallback) Resolution 720 (Tag) Revision v1 from FileName
2022-03-14 18:01:34.2|Trace|AggregateQuality|Considering Source Television (Fallback) Resolution 720 (Tag) Revision v1 from DownloadClientItem
2022-03-14 18:01:34.2|Trace|AugmentQualityFromMediaInfo|Resolution 1280x720 considered 720p
2022-03-14 18:01:34.2|Trace|AggregateQuality|Considering Source Unknown (Default) Resolution 720 (MediaInfo) Revision  from MediaInfo
2022-03-14 18:01:34.2|Trace|AggregateQuality|Selected Source Television (Fallback) Resolution 720 (MediaInfo) Revision v1
2022-03-14 18:01:34.2|Debug|AggregateQuality|Using quality: HDTV-720p v1
2022-03-14 18:01:34.2|Debug|AbsoluteEpisodeNumberSpecification|Series type is not Anime, skipping check
2022-03-14 18:01:34.2|Trace|AlreadyImportedSpecification|Episode file has not been imported
2022-03-14 18:01:34.2|Trace|ConfigService|Using default config value for 'episodetitlerequired' defaultValue:'Always'
2022-03-14 18:01:34.2|Debug|EpisodeTitleSpecification|Episode aired more than 1 day ago
2022-03-14 18:01:34.2|Trace|ConfigService|Using default config value for 'skipfreespacecheckwhenimporting' defaultValue:'False'
2022-03-14 18:01:34.2|Trace|ConfigService|Using default config value for 'minimumfreespacewhenimporting' defaultValue:'100'
2022-03-14 18:01:34.2|Debug|MatchesFolderSpecification|No folder ParsedEpisodeInfo, skipping check
2022-03-14 18:01:34.2|Debug|VideoFileInfoReader|Getting media info from \\1.1.1.1\downloads\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.2|Trace|MediaInfo|Read file offset 0-5898240 (5898240 bytes)
2022-03-14 18:01:34.2|Trace|MediaInfo|Read file offset 745804813-745826220 (21407 bytes)
2022-03-14 18:01:34.2|Trace|MediaInfo|Read a total of 5919647 bytes (0.8%)
2022-03-14 18:01:34.2|Debug|DetectSample|Runtime is over 90 seconds
2022-03-14 18:01:34.2|Trace|ConfigService|Using default config value for 'downloadclientworkingfolders' defaultValue:'_UNPACK_|_FAILED_'
2022-03-14 18:01:34.2|Trace|ConfigService|Using default config value for 'downloadpropersandrepacks' defaultValue:'PreferAndUpgrade'
2022-03-14 18:01:34.2|Trace|EpisodeFilePreferredWordCalculator|No stored scene name for [460] Season 4\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.2|Trace|EpisodeFilePreferredWordCalculator|No stored scene name for [460] Season 4\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:34.2|Trace|EpisodeFilePreferredWordCalculator|Calculating preferred word score for 'Season 4\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv'
2022-03-14 18:01:34.2|Trace|EpisodeFilePreferredWordCalculator|Calculated preferred word score for 'Season 4\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv': 0 (0 match(es))
2022-03-14 18:01:34.2|Debug|ImportDecisionMaker|File accepted
2022-03-14 18:01:34.2|Trace|Http|Res: 2236 [GET] /api/v3/manualimport?downloadId=FFEF1D8B1B7ACB3D57FD5D6FAF17249960B596A4&folder=&filterExistingFiles=true: 200.OK (284 ms)
2022-03-14 18:01:34.2|Debug|Api|[GET] /api/v3/manualimport?downloadId=FFEF1D8B1B7ACB3D57FD5D6FAF17249960B596A4&folder=&filterExistingFiles=true: 200.OK (284 ms)
2022-03-14 18:01:35.2|Trace|Http|Req: 2237 [POST] /api/v3/command (from ::1 Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36 Edg/99.0.1150.39)
2022-03-14 18:01:35.2|Trace|CommandQueueManager|Publishing ManualImport
2022-03-14 18:01:35.2|Trace|CommandQueueManager|Checking if command is queued or started: ManualImport
2022-03-14 18:01:35.2|Trace|CommandQueueManager|Inserting new command: ManualImport
2022-03-14 18:01:35.2|Trace|CommandExecutor|ManualImportCommand -> ManualImportService
2022-03-14 18:01:35.2|Trace|CommandQueueManager|Marking command as started: ManualImport
2022-03-14 18:01:35.2|Trace|Http|Res: 2237 [POST] /api/v3/command: 201.Created (5 ms)
2022-03-14 18:01:35.2|Debug|Api|[POST] /api/v3/command: 201.Created (5 ms)
2022-03-14 18:01:35.2|Trace|EventAggregator|Publishing CommandUpdatedEvent
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.2|Trace|ManualImportService|Manually importing 1 files using mode Auto
2022-03-14 18:01:35.2|Trace|EventAggregator|Publishing CommandUpdatedEvent
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.2|Trace|ManualImportService|Processing file 1 of 1
2022-03-14 18:01:35.2|Trace|EventAggregator|Publishing CommandUpdatedEvent
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.2|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.2|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv'
2022-03-14 18:01:35.2|Trace|Parser|^\[(?<subgroup>.+?)\][-_. ]?(?<title>[^-]+?)(?:(?<![-_. ]|\b[0]\d+) - )(?:[-_. ]?(?<absoluteepisode>\d{2,3}(\.\d{1,2})?(?!\d+)))+(?:[-_. ]+(?<special>special|ova|ovd))?.*?(?<hash>\[\w{8}\])?(?:$|\.mkv)
2022-03-14 18:01:35.2|Debug|Parser|Episode Parsed. Shingeki no Kyojin (The Final Season) - 085 
2022-03-14 18:01:35.2|Debug|Parser|Language parsed: English
2022-03-14 18:01:35.2|Debug|QualityParser|Trying to parse quality for [SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:35.2|Debug|Parser|Quality parsed: HDTV-720p v1
2022-03-14 18:01:35.2|Debug|Parser|Release Group parsed: SubsPlease
2022-03-14 18:01:35.2|Debug|Parser|Release Hash parsed: 43E56853
2022-03-14 18:01:35.2|Debug|VideoFileInfoReader|Getting media info from \\1.1.1.1\downloads\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv
2022-03-14 18:01:35.2|Trace|MediaInfo|Read file offset 0-5898240 (5898240 bytes)
2022-03-14 18:01:35.2|Trace|MediaInfo|Read file offset 745804813-745826220 (21407 bytes)
2022-03-14 18:01:35.2|Trace|MediaInfo|Read a total of 5919647 bytes (0.8%)
2022-03-14 18:01:35.3|Debug|Parser|Parsing string '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]'
2022-03-14 18:01:35.3|Debug|SceneMappingService|Found scene mapping for: Shingeki no Kyojin (The Final Season). TVDB ID for mapping: 267440
2022-03-14 18:01:35.3|Debug|ParsingService|Using absolute episode number 85 for: Attack on Titan - TVDB: 4x26
2022-03-14 18:01:35.3|Debug|AggregateLanguage|Using language: English
2022-03-14 18:01:35.3|Trace|PreferredWordService|Calculating preferred word score for '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]'
2022-03-14 18:01:35.3|Trace|PreferredWordService|Calculated preferred word score for '[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]': 0 (0 match(es))
2022-03-14 18:01:35.3|Trace|AggregateQuality|Considering Source Television (Fallback) Resolution 720 (Tag) Revision v1 from FileName
2022-03-14 18:01:35.3|Trace|AggregateQuality|Considering Source Television (Fallback) Resolution 720 (Tag) Revision v1 from DownloadClientItem
2022-03-14 18:01:35.3|Trace|AugmentQualityFromMediaInfo|Resolution 1280x720 considered 720p
2022-03-14 18:01:35.3|Trace|AggregateQuality|Considering Source Unknown (Default) Resolution 720 (MediaInfo) Revision  from MediaInfo
2022-03-14 18:01:35.3|Trace|AggregateQuality|Selected Source Television (Fallback) Resolution 720 (MediaInfo) Revision v1
2022-03-14 18:01:35.3|Debug|AggregateQuality|Using quality: HDTV-720p v1
2022-03-14 18:01:35.3|Warn|ImportApprovedEpisodes|Couldn't import episode \\1.1.1.1\downloads\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv

[v3.0.7.1477] System.ArgumentException: The UNC path should be of the form \\server\share.
   at System.IO.LongPathHelper.Normalize(String path, UInt32 maxPathLength, Boolean checkInvalidCharacters, Boolean expandShortPaths)
   at System.IO.Path.NewNormalizePath(String path, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.GetFullPathInternal(String path)
   at System.IO.DirectoryInfo.Init(String path, Boolean checkHost)
   at System.IO.DirectoryInfo..ctor(String path)
   at NzbDrone.Common.Extensions.PathExtensions.IsParentPath(String parentPath, String childPath) in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Common\Extensions\PathExtensions.cs:line 116
   at NzbDrone.Core.MediaFiles.EpisodeImport.ImportApprovedEpisodes.GetOriginalFilePath(DownloadClientItem downloadClientItem, LocalEpisode localEpisode) in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\MediaFiles\EpisodeImport\ImportApprovedEpisodes.cs:line 169
   at NzbDrone.Core.MediaFiles.EpisodeImport.ImportApprovedEpisodes.Import(List`1 decisions, Boolean newDownload, DownloadClientItem downloadClientItem, ImportMode importMode) in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\MediaFiles\EpisodeImport\ImportApprovedEpisodes.cs:line 107

2022-03-14 18:01:35.3|Trace|ManualImportService|Manually imported 1 files
2022-03-14 18:01:35.3|Trace|EventAggregator|Publishing CommandUpdatedEvent
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.3|Trace|CommandQueueManager|Updating command status
2022-03-14 18:01:35.3|Trace|EventAggregator|Publishing CommandUpdatedEvent
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandUpdatedEvent -> CommandModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandUpdatedEvent <- CommandModule
2022-03-14 18:01:35.3|Trace|EventAggregator|Publishing CommandExecutedEvent
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandExecutedEvent -> TaskManager
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandExecutedEvent <- TaskManager
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandExecutedEvent -> TaskModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandExecutedEvent <- TaskModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandExecutedEvent -> TaskModule
2022-03-14 18:01:35.3|Trace|EventAggregator|CommandExecutedEvent <- TaskModule
2022-03-14 18:01:35.3|Trace|CommandExecutor|ManualImportCommand <- ManualImportService [00:00:00.0980558]
```

### Anything else?

I have setup a development environment to dig a little deeper and found the following.
Here is a code snippet of where the error occurs, including objects at the time of execution.
```cs
/*******
THE BELOW IS AN EXAMPLE OF THE PROGRAM STATE AT THE TIME OF EXECUTION
VARIABLES HAVE BEEN COMMENTED IN FOR READBILITY
*******/

/*******
downloadClientItem object:
"Flood":"[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv"
    CanBeRemoved: false
    CanMoveFiles: false
    Category: "tv-son"
    DownloadClientInfo: {NzbDrone.Core.Download.DownloadClientItemClientInfo}
    DownloadId: "FFEF1D8B1B7ACB3D57FD5D6FAF17249960B596A4"
    IsEncrypted: false
    Message: ""
    OutputPath: {\\1.1.1.1\downloads\}
    RemainingSize: 0
    RemainingTime: null
    Removed: false
    SeedRatio: 0.004999999888241291
    Status: Completed
    Title: "[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv"
    TotalSize: 745826220

downloadClientItem.DownloadClientInfo property:
{NzbDrone.Core.Download.DownloadClientItemClientInfo}
    Id: 1
    Name: "Flood"
    Protocol: Torrent
    Type: "Flood"
*******/

/*******
localEpisode object:
{\\1.1.1.1\downloads\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv}
    DownloadClientEpisodeInfo: {Shingeki no Kyojin (The Final Season) - 085 HDTV-720p v1}
    Episodes: Count = 1
    ExistingFile: false
    FileEpisodeInfo: {Shingeki no Kyojin (The Final Season) - 085 HDTV-720p v1}
    FolderEpisodeInfo: null
    IsSpecial: false
    Language: {English}
    MediaInfo: {NzbDrone.Core.MediaFiles.MediaInfo.MediaInfoModel}
    OtherVideoFiles: false
    Path: "\\\\1.1.1.1\\downloads\\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv"
    PreferredWordScore: 0
    Quality: {HDTV-720p v1}
    ReleaseGroup: "SubsPlease"
    SceneName: "[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853]"
    SceneSource: true
    SeasonNumber: 4
    Series: {Name = "Series" FullName = "NzbDrone.Core.Tv.Series"} ID = 3
    Size: 745826220
*******/

namespace NzbDrone.Core.MediaFiles.EpisodeImport
{
    ...snip...

    public class ImportApprovedEpisodes : IImportApprovedEpisodes
    {

        ...snip...

        private string GetOriginalFilePath(DownloadClientItem downloadClientItem, LocalEpisode localEpisode)
        {
            var path = localEpisode.Path; 
            // var path = "\\1.1.1.1\downloads\[SubsPlease] Shingeki no Kyojin (The Final Season) - 85 (720p) [43E56853].mkv"

            if (downloadClientItem != null && !downloadClientItem.OutputPath.IsEmpty)
            {
                var outputDirectory = downloadClientItem.OutputPath.Directory.ToString();
                // downloadClientItem.OutputPath = "\\1.1.1.1\downloads\"
                // downloadClientItem.OutputPath.Directory = "\\1.1.1.1\"
                // var outputDirectory = "\\1.1.1.1\"

                                       //  |---- The exception is thrown here by 'IsParentPath' method. Because \\1.1.1.1\ is not a valid UNC path.
                if (outputDirectory.IsParentPath(path))
                {
                    return outputDirectory.GetRelativePath(path);
                }
            }

            ...snip...

        }

    ...snip...

    }
}
```
After review further up the stack, I can see that Flood handles the property `DownloadClientItem.OutputPath` differently than other clients - Deluge for example, puts the torrent name at the end.

In the case of Flood, it omits the torrent name for `DownloadClientItem.OutputPath` and instead uses a `Flood.GetImportItem(DownloadClientItem, DownloadClientItem)` method to determine the output path of a file, I think?

This could be combined with the way `OsPath.OutputPath.Directory` is handling UNC paths, given other Download Clients use a full path of `\\1.1.1.1\path\episode.mkv` and Flood does not?

I am only be guessing, I don't have the familiarity with the code base to suggest fixes or really understand the problem with the short amount of time I have spent looking into this.
Hopefully this helps you in tracking down this issue. 
Also my development environment is still setup if you need further information.

**Comment by manniebugs** [comment]
Update (with workaround):

**TLDR (workaround):** Use `\\?\UNC\1.1.1.1\subfolder` instead of `\\1.1.1.1\subfolder\` for `Settings > Download Client > Remote Path Mappings > Local Path`

## Why that works
I found some more time to do more digging, I setup a Radarr environment to see how it works.

In Sonarr:
```cs
namespace NzbDrone.Common.Extensions
{
    public static class PathExtensions
    {
        ...snip...

        public static bool IsParentPath(this string parentPath, string childPath)
        {
            ...snip...
           
            // 'DirectoryInfo(string)' is the method throwing the exception about the UNC path.
            // Radarr is using .net 6 version of the library (System.Runtime.dll)
            // Sonarr is using the .net 4.6.2 version (mscorlib.dll)
            // Looked into it further and the .net 4.6.2 does not support '\\server' URI's w/o a subdirectory
            // https://stackoverflow.com/questions/3567063/get-a-list-of-all-unc-shared-folders-on-a-local-network-server
            var parent = new DirectoryInfo(parentPath); 
            var child = new DirectoryInfo(childPath);

            ...snip...
        }

    ...snip...
    }
}
```

After some digging I found this gem [File path formats on Windows systems: DOS device paths](https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats#dos-device-paths)
Which states 
> DOS device path syntax is supported on .NET implementations running on Windows starting with .NET Core 1.1 and .NET Framework 4.6.2.

Which allows:
>There is a specific link for UNCs that is called, not surprisingly, UNC. For example:
>`\\.\UNC\Server\Share\Test\Foo.txt` or `\\?\UNC\Server\Share\Test\Foo.txt`

Changing the `Settings > Download Client > Remote Path Mappings > Local Path` to `\\?\UNC\10.0.50.2\downloads` **fixes the issue**.

I'm going to leave this issue open in case you want to add it to documentation or something. Otherwise feel free to close.
